### ISSUE TITLE
Create web-based version of StormSurgeViz using WMS services and CSW catalog

### ISSUE THREAD

**Issue by rsignell-usgs** [neracoos-open/NFWF#7]
Under National Hurricane Center funding, @BrianOBlanton created the Matlab-based StormSurgeViz application http://renci-unc.github.io/StormSurgeViz/ for visualizing UGRID compliant data available via OPeNDAP. 

I spoke with Brian on Friday, Sep 9, and he was very interested in the possibility of us developing a web-based version of this tool that would use WMS services in the browser instead of downloading data into Matlab using OPeNDAP.   

Currently the tool also uses just a static list of datasets (OPeNDAP endpoints), and Brian said a catalog service that would be easy to maintain and have folks contribute to would be fantastic. 

The WMS services for UGRID could either come from ncWMS2 (included with TDS 5.0) or via sci-wms.  We know there are issues with both ncWMS2 and sci-wms delivering WMS from large UGRID datasets, and we need to do some testing to understand the performance issues.   

Sci-wms is python-based, and python needs a common API to work with CF, UGRID and SGRID datasets.     

So a list of useful tasks for NFWF to contribute to the inundation forecast community could be:
- [x] Evaluate performance of WMS from both ncWMS2 and sci-wms using both RENCI and ESTOFS UGRID forecast datasets
- [x] Seek performance enhancements for WMS from UGRID models
- [x] Built out nascent `gridded` package at https://github.com/pyoceans/gridded to provide uniform support for CF, UGRID and SGRID model datasets.  
- [ ] Establish pycsw catalog service for inundation community, with python scripts that crawl thredds catalog endpoints on a github repository so that new catalogs and filtering operations on those catalogs can be modified via pull requests.  
- [ ] Build web apps that consumes WMS endpoints and has a dynamic catalog controlled by CSW, using both Axiom infrastructure and TerriaJS. 

Let's discuss, and then turn these into issues and milestones when we have agreed on a plan.

**Comment by rsignell-usgs** [comment]
@cbcunc, have you made any progress here?   
For example, do you now have WMS services enabled for RENCI UGRID runs via ncWMS2 or sci-wms?

**Comment by cbcunc** [comment]
No, I shifted efforts to pycsw where I'm having more success before returning to sciwms. I could give ncwms2 a try. I've largely been reassigned to other things so this may take awhile, as it already has. Currently also working on Hyrax issues.

**Comment by rsignell-usgs** [comment]
@cbcunc , what is your pycsw CSW endpoint?

**Comment by cbcunc** [comment]
http://csw.renci.org:8000/ . As you can see, it has no contents. Back in March I was working on a Thredds harvester and metadata ingestor for it based on the IOOS harvester. However, i was completely reassigned to other work before that got farther than watching the IOOS harvester fall over when presented with a real life catalog. I don't get to do marine work at all at RENCI anymore, having been reassigned to devops for other research projects.
